Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/BlogRates/DTOs/Validators/BlogRateDtoValidator.cs

```diff
@@ -0,0 +1,25 @@
+﻿using BlogApp.Application.Contracts.Persistence;
+using FluentValidation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BlogApp.Application.Features.BlogRates.DTOs.Validators
+{
+    public class BlogRateDtoValidator : AbstractValidator<BlogRateDto>
+    {
+        public BlogRateDtoValidator(IUnitOfWork unitOfWork) {
+            Include(new IBlogRateDtoValidator());
+            
+            RuleFor(p => p.BlogId)
+            .GreaterThan(0)
+            .MustAsync(async (id, token) => await unitOfWork.BlogRateRepository.BlogExists(id)).WithMessage($"Blog not found");
+            RuleFor(p => p.RaterId)
+            .GreaterThan(0)
+            .MustAsync(async (id, token) => await unitOfWork.BlogRateRepository.RaterExists(id)).WithMessage($"Rater not found");
```

Comment:
I think you added that checking if the rat**er** is exist or not but I was saying that checking the existence of the **rate** itself is necessary because if you are going to update it you have to check if it exists or not.  

this is an example they used for updating comment

RuleFor(p => p.Id)
            .GreaterThan(0)
            .MustAsync(async (id, token) => await unitOfWork._CommentRepository.Exists(id)).WithMessage($"Comment not found");